src/lib/container.interface.ts

```diff
@@ -6,6 +6,8 @@ export interface IContainerOptions {
 }
 
 export interface IContainer {
+    [key: string]: any;
```

Comment:
Doing that it's a bad practise. Interface (aka super type) should define only public methods. Unless you use interface for defining structure of value objects. Please remove that.